ansible_base/lib/utils/validation.py

```
@@ -17,6 +21,112 @@
+
+
+def _normalize_for_markup_compare(text):
+    """Undo the transformations nh3 applies that are *not* markup removal, so
+    only genuine tag/attribute stripping causes a mismatch: entity escaping
+    (``&`` -> ``&amp;``) and CRLF/CR -> LF newline normalization."""
+    return html_mod.unescape(text).replace('\r\n', '\n').replace('\r', '\n')
+
+
+def _markup_variants(text):
+    """Yield ``text`` plus a normalized form that defeats common tag
+    obfuscation: NFKC folds fullwidth brackets (``＜`` -> ``<``) and
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `＜` (FULLWIDTH LESS-THAN SIGN). Did you mean `<` (LESS-THAN SIGN)?

(RUF002)